server/protocols/webadmin/webadmin-mailqueue/src/test/java/org/apache/james/webadmin/routes/MailQueueRoutesTest.java

```diff
@@ -513,6 +514,124 @@ public void forcingDelayedMailsDeliveryShouldRejectRequestWithoutBody() throws E
 
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_1)
+            .sender(SENDER_1_JAMES_ORG)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_2)
+            .sender(SENDER_2_JAMES_ORG)
+            .build());
+
+        String taskId = with()
+            .param("sender", SENDER_1_JAMES_ORG)
+            .delete(FIRST_QUEUE + "/mails")
+            .jsonPath()
+            .getString("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"));
+
+        assertThat(queue.browse())
+            .hasSize(1)
+            .first()
+            .satisfies(mailView -> {
+                assertThat(mailView.getMail().getName()).isEqualTo(FAKE_MAIL_NAME_2);
+                assertThat(mailView.getMail().getSender().asString()).isEqualTo(SENDER_2_JAMES_ORG);
+            });
+
+    }
+
+    @Test
+    public void deleteMailsShouldDeleteMailsWhenNameIsGiven() throws Exception {
+        MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_1)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_2)
+            .build());
+
+        String taskId = with()
+            .param("name", FAKE_MAIL_NAME_1)
+            .delete(FIRST_QUEUE + "/mails")
+            .jsonPath()
+            .getString("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"));
+
+        assertThat(queue.browse())
+            .hasSize(1)
+            .first()
+            .satisfies(mailView -> assertThat(mailView.getMail().getName()).isEqualTo(FAKE_MAIL_NAME_2));
+    }
+
+    @Test
+    public void deleteMailsShouldDeleteMailsWhenRecipientIsGiven() throws Exception {
+        MemoryMailQueue queue = mailQueueFactory.createQueue(FIRST_QUEUE);
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_1)
+            .recipient(RECIPIENT_JAMES_ORG)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_2)
+            .recipient(RECIPIENT_1_JAMES_ORG)
+            .build());
+
+        queue.enQueue(FakeMail.builder()
+            .name(FAKE_MAIL_NAME_3)
+            .recipient(RECIPIENT_2_JAMES_ORG)
+            .build());
+
+        String taskId = with()
+            .param("recipient", RECIPIENT_JAMES_ORG)
+            .delete(FIRST_QUEUE + "/mails")
+            .jsonPath()
+            .getString("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"));
+
+        assertThat(queue.browse()).hasSize(2);
```

Comment:
You can be doing:

```
assertThat(X)
    .hasSize(2)
    .anySatisfy(...)
```

Here also `anySatisfy` is too verbose. I suggest you get a look at `assertThat::extract` for a lighter syntax:

```
assertThat(queue.browse())
   .hasSize(2)
   .extract(MailItem::getMail)
   .extract(Mail::getName)
   .containsOnly(Fake_MAIL_NAME_2, Fake_MAIL_NAME_3);
```

You can also restrict your asserts to the mail name. That would seem enough to me.